templates/tfengine/components/resources/compute_instance_templates/main.tf

```diff
@@ -0,0 +1,67 @@
+{{- /* Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */ -}}
+
+{{range .compute_instance_templates -}}
+{{- $template_resource_name := resourceName . "name_prefix" }}
+{{- $network_project_id := get . "network_project_id" "${var.project_id}"}}
+{{- $subnet := .subnet}}
+module "{{$template_resource_name}}" {
+  source  = "terraform-google-modules/vm/google//modules/instance_template"
+  version = "~> 3.0.0"
+
+  name_prefix        = "{{.name_prefix}}"
+  project_id         = var.project_id
+  region             = "{{get . "compute_region" $.compute_region}}"
+  subnetwork_project = "{{$network_project_id}}"
+  subnetwork         = "{{$subnet}}"
+
+  {{- hclField . "machine_type"}}
+  {{- hclField . "image_family"}}
+  {{- hclField . "image_project"}}
+  {{- hclField . "disk_size_gb"}}
+  {{- hclField . "disk_type"}}
+  {{- hclField . "preemptible"}}
+
+  service_account = {
+    email = "{{.service_account}}"
```

Comment:
is there a way to auto-allocated default SA?

Reply:
It would become the default compute SA, so I think passing an SA should be a good best practice. It's also what's recommended here: https://cloud.google.com/compute/docs/access/service-accounts#associating_a_service_account_to_an_instance